sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -934,6 +934,23 @@ object SQLConf {
     .intConf
     .createWithDefault(4096)
 
+   val PARQUET_FIELD_ID_ENABLED =
+    buildConf("spark.sql.parquet.fieldId.enabled")
+      .doc("Field ID is a native field of the Parquet schema spec. When enabled, Parquet readers" +
+        " will use field IDs (if present) in the requested Spark schema to look up Parquet" +
+        " fields instead of using column names; Parquet writers will also populate the field Id" +
+        " metadata (if present) in the Spark schema to the Parquet schema.")
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
I am wondering if there should be two flags or one flag for reads only that is switched to false to enable easier migration.

Reply:
I am inclined to keep this config as false by default

Reply:
You meant one flag for read and another flag for write?

Reply:
I was thinking about having 2 flags:
- writing field ids, true by default
- reading field ids, false by default initially to avoid any regressions.

Alternatively, we can always write field ids, but reads need to be configurable.